File: lib/src/main/java/com/sd/lib/http/impl/GetRequest.java
package com.sd.lib.http.impl;

import com.sd.lib.http.IGetRequest;
import com.sd.lib.http.IResponse;

public class GetRequest extends BaseRequest implements IGetRequest
{
    @Override
    protected IResponse doExecute() throws Exception
    {
        HttpRequest request = newHttpRequest(HttpRequest.append(getUrl(), getParams().toMap()), HttpRequest.METHOD_GET);
        return new Response(request);
    }
}

File: lib/src/main/java/com/sd/lib/http/impl/GetRequest.kt
package com.sd.lib.http.impl

import com.sd.lib.http.IGetRequest
import com.sd.lib.http.IResponse
import com.sd.lib.http.impl.HttpRequest

class GetRequest : BaseRequestImpl(), IGetRequest {
    override fun doExecute(): IResponse {
        val getUrl = HttpRequest.append(url, params.toMap())
        val request = newHttpRequest(getUrl, HttpRequest.METHOD_GET)
        return Response(request)
    }
}